feat(i18n): Add Farsi (Persian) localization file

rafi-ghanbari opened the pull request:
### Summary
This Pull Request introduces the Farsi/Persian language localization file (`persian.json`) to the project. 

### Detailed Changes
* **New File:** `persian.json` is added containing translations for all key strings found in `english.json`.
* **Language Scope:** The translation covers all main sections, including 'Connecting', 'Error', 'Main', and 'Patches'.
* The `LanguageName` is set to "فارسی".

rafi-ghanbari committed:
feat(i18n): Add Farsi (Persian) localization file

exelix11 reviewed (commented):
Did you test this on console? 
For other languages that don't use the latin script we need a different font and glyph ranges, for example see the chinese files. 
I think this file as-is will fail to load the text only showing '?' characters.

rafi-ghanbari committed:
Add Farsi/Persian support with NotoSansArabic font

rafi-ghanbari commented:
Thanks for the feedback! You were right about the font support.

I have updated the PR to fully support the script:

1. I added NotoSansArabic-Regular.ttf to the project.
2. I defined a new Arabic glyph range in the C++ code to ensure characters load correctly (similar to how the Chinese implementation works).
3. I updated the `persian.json` config to point to this new font and range.

I haven't tested this on a physical console yet, but I've matched the implementation logic used for other non-Latin languages in the project.

rafi-ghanbari commented:
Im well aware that the Nintendo Switch doesn't natively support Farsi. However, I wanted to explain that in Iran, most consoles are homebrewed (jailbroken ) to allow for homebrew apps, custom settings, and more. My intention was to provide a better experience for that specific community.

That being said I completely understand if adding languages not officially supported by the OS is outside the scope of your project .

I apologize for not providing enough context initially, which led to the misunderstanding that this was an ai generated request. I didn't mean to waste your time.
